spring-cloud-aws-s3-parent/spring-cloud-aws-s3/src/main/java/io/awspring/cloud/s3/S3Template.java

```diff
@@ -128,4 +137,34 @@ public S3Resource download(String bucketName, String key) {
 		return new S3Resource(bucketName, key, s3Client, s3OutputStreamProvider);
 	}
 
+	@Override
+	public URL createSignedGetURL(String bucketName, String key, int durationMinutes){
+		S3Presigner presigner = S3Presigner.builder().build();
```

Comment:
`S3Presigner` should be a bean created in `S3AutoConfiguration`. It should have `@ConditionalOnMissingBean` so that users can overwrite it, and should take `S3Configuration` from `S3Properties`.